client/src/plugins/privacy-preferences/PrivacyPreferencesView.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import React, { PureComponent } from 'react';
+
+import {
+  Modal
+} from '../../app/primitives';
+
+import css from './PrivacyPreferencesView.less';
+
+import {
+  PRIVACY_TEXT_FIELD,
+  PRIVACY_POLICY_URL,
+  LEARN_MORE_TEXT,
+  PRIVACY_POLICY_TEXT,
+  PREFERENCES_LIST,
+  OK_BUTTON_TEXT,
+  CANCEL_BUTTON_TEXT,
+  TITLE,
+  DEFAULT_VALUES
+} from './constants';
+
+class PrivacyPreferencesView extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = { ...props.preferences };
+  }
+
+  isEnabled = (key) => {
+    const {
+      preferences
+    } = this.props;
+
+    if (!preferences) {
+      return DEFAULT_VALUES[key];
+    }
+
+    return preferences[key];
+  }
+
+  renderPreferences() {
+    return PREFERENCES_LIST.map((item) => (
+      <div key={ item.key } className='grid-container'>
+        <div className='grid-item item1'>
+          <input
+            type="checkbox"
+            defaultChecked={ this.isEnabled(item.key) }
+            onChange={ (event) => {
+              this.setState({ [item.key]: event.target.checked });
+            } } />
+        </div>
+        <div className='grid-item item2'>
+          <div className="checkboxLabel"> { item.title } </div>
+          <div className="checkboxExplanation"> { item.explanation } </div>
+        </div>
+      </div>
+    ));
+  }
+
+  render() {
+
+    const {
+      onClose,
+      onSaveAndClose,
+      hasCancel
+    } = this.props;
+
+    return (
+      <Modal className={ css.View } hideCloseIcon={ true }>
```

Comment:
This is the place where I'd assume => no `onClose` => no close button being shown.

`onClose` passed => standard modal interactions work => `ESC`, backdrop click

Reply:
This is a nice solution, however Modal.js has a default value for onClose: https://github.com/camunda/camunda-modeler/blob/c27d749c449233e8c82f42f232f9d2cf70f6a237/client/src/app/primitives/modal/Modal.js#L79

So we cannot get away with `if(!this.props.onClose)`, we can compare the function with noop function though, but I'm not exactly sure if that'd be semantically correct.

So we should either remove the default value or pass the hideCloseIcon parameter, I'd say lets keep the hideCloseIcon parameter. What do you say?

Reply:
Let's not think on the level as low as "our code is like this, we cannot change it" here. The fact that there was a noop in the past should not keep us from having a better API in the future.

What do we loose by making showing the close icon dependent on the onClose property? I guess not much. Showing a close button when the modal is not closable (and won't close) is a bug, not a feature.

On the other hand, in terms of API of our Modal: Should we as developers take two things into account (hideCloseButton, not adding onClick) to make it a confirmation dialog?